discovery_child_development/analysis/baseline_model.py

```diff
@@ -0,0 +1,423 @@
+
+    if targets is not None:
+        missing_labels = pd.Series([], index=[])
+        for x in targets:
+            if x not in label_probabilities.index:
+                missing_labels = missing_labels.append(pd.Series([float(0)], index=[x]))
+        label_probabilities = label_probabilities.append(missing_labels)
+
+    return label_probabilities
+
+
+def run_baseline_model(
+    model_type: str = "majority_combination",
+    wandb_run: bool = True,
+    score_threshold=0.3,
+    s3_bucket: str = S3_BUCKET,
+    input_path: str = INPUT_PATH,
+    concept_ids: str = CONCEPT_IDS,
+    vectors_filepath: str = VECTORS_FILEPATH,
+    model_path: str = MODEL_PATH,
+):
+    valid_inputs = ["majority_combination", "most_probable"]  # "majority_label",
+
+    if model_type not in valid_inputs:
+        raise ValueError(
+            f"Invalid input. Expected one of {valid_inputs}, got '{model_type}'"
+        )
+
+    # Load the data
+    training_data_filename = (
+        f"openalex_data_{concept_ids}_year-2019-2020-2021-2022-2023_train.csv"
+    )
+
+    openalex_data = S3.download_obj(
+        s3_bucket,
+        path_from=f"{input_path}{training_data_filename}",
+        download_as="dataframe",
+        kwargs_reading={"index_col": 0},
+    )
+
+    if wandb_run:
+        # Initialize a wandb run and log score threshold with wandb
+        run = wandb.init(
+            project="ISS supervised ML",
+            job_type="Baseline modeling",
+            save_code=True,
+            config={
+                "baseline_model_type": model_type,
+                "score_threshold": score_threshold,
+            },
+        )
+        # We will use set the wandb description to be the dataset name (which includes details of concepts and years)
+        # - we set the description, and not the name, as there is a limit on the length of artifact names.
+        # wandb artifact names also cannot include "|" so we replace this.
+        wandb_name = training_data_filename.replace("|", "_")
+        # add reference to this data in wandb
+        wb.add_ref_to_data(
+            run,
+            "openalex_train_data_raw",
+            wandb_name,
+            S3_BUCKET,
+            training_data_filename,
+        )
+
+    # Filter the data using a score threshold (0.3 is the threshold used by OpenAlex)
+    openalex_data_wide = (
+        openalex_data[openalex_data["score"] >= score_threshold]
+        # Squash sub-categories into one tuple per work (rather than one row per sub-category per work)
+        .groupby(["openalex_id", "text"])["sub_category"]
+        .agg(lambda x: tuple(set(x)))
+        .reset_index()
+    )
+    # Set the index - useful later for creating training/validation split
+    openalex_data_wide = openalex_data_wide.set_index("openalex_id")
+
+    top_combinations, most_common_combination = find_most_frequent_labels(
+        openalex_data_wide, label_col="sub_category", head=20
+    )
```

Comment:
Are you using the `most_common_combination` variable here later on in the function - if not, can replace with `_`

Also, I'd call this function where it is needed, ie after you've checked `if model_type == "majority_combination":`

Reply:
🤯 thanks, yes I can use _ instead!

And good spot - it is indeed only needed for the "majority_combination" type